unblob/handlers/compression/lzh.py

```diff
@@ -0,0 +1,102 @@
+import io
+from typing import List, Optional
+
+from structlog import get_logger
+
+from ...file_utils import Endian
+from ...models import StructHandler, ValidChunk
+
+logger = get_logger()
+
+PADDING_LEN = 2
+LEVEL_IDENTIFIERS = [0, 1, 2]
+METHOD_IDS = [
+    b"-lh0-",
+    b"-lzs-",
+    b"-lz4-",
+    b"-lh1-",
+    b"-lh2-",
+    b"-lh3-",
+    b"-lh4-",
+    b"-lh5-",
+    b"-lh6-",
+    b"-lh7-",
+    b"-lh8-",
+    b"-lhd-",
+]
+
+
+class LZHHandler(StructHandler):
+
+    NAME = "lzh"
+
+    YARA_RULE = r"""
+        strings:
+            // 2 header bytes followed by method ID
+            $magic = /[\x00-\xff][\x00-\xff](-lh0-|-lzs-|-lz4-|-lh1-|-lh2-|-lh3-|-lh4-|-lh5-|-lh6-|-lh7-|-lh8-|-lhd-)/
+        condition:
+            $magic
+    """
+
+    C_DEFINITIONS = r"""
+        struct lzh_default_header {
+            uint8 header_size;          // excludes extended headers size
+            uint8 header_checksum;
+            char method_id[5];
+            uint32 compressed_size;     // includes all extended headers size (if level 1)
+            uint32 uncompressed_size;
+            uint32 timestamp;
+            uint8 fd_attribute;
+            uint8 level_identifier;
+        };
+
+        struct level_2_header {
+            uint16 header_size; // includes all extended headers
+            char method_id[5];
+            uint32 compressed_size;     // excludes all extended headers
+            uint32 uncompressed_size;
+            uint32 timestamp;
+            uint8 fd_attribute;
+            uint8 level_identifier;
+        };
+    """
+    HEADER_STRUCT = "lzh_default_header"
+
+    def calculate_chunk(
+        self, file: io.BufferedIOBase, start_offset: int
+    ) -> Optional[ValidChunk]:
+
+        header = self.parse_header(file, Endian.LITTLE)
+        file.seek(-len(header), io.SEEK_CUR)
+
+        if (
+            header.method_id not in METHOD_IDS
+            or header.level_identifier not in LEVEL_IDENTIFIERS
+        ):
+            return
+
+        if header.level_identifier == 0x2:
+            # magic trick, just lookup the header definition of level 2
```

Comment:
Is there any official header definition? Maybe the one you [linked in the description of the PR](https://web.archive.org/web/20021005080911/http://www.osirusoft.com/joejared/lzhformat.html)?
If so, you might want to add here in the comment as well since you are referring it.

Reply:
Explained the magic trick with an inline comment and referenced the level 2 definition I kept in `C_DEFINITIONS` for documentation purposes.